api/api/versions_api.go

```diff
@@ -100,17 +99,22 @@ func (c *VersionsController) ListVersions(r *http.Request, vars map[string]strin
 	return OkWithHeaders(versions, responseHeaders)
 }
 
-func (c *VersionsController) CreateVersion(r *http.Request, vars map[string]string, _ interface{}) *Response {
+func (c *VersionsController) CreateVersion(r *http.Request, vars map[string]string, body interface{}) *Response {
 	ctx := r.Context()
 
+	versionPost, ok := body.(*models.VersionPost)
+	if !ok {
+		return BadRequest("Unable to parse request body")
+	}
+
 	modelID, _ := models.ParseID(vars["model_id"])
 
 	model, err := c.ModelsService.FindByID(ctx, modelID)
 	if err != nil {
 		return NotFound(fmt.Sprintf("Model with given `model_id: %d` not found", modelID))
 	}
 
-	mlflowClient := mlflow.NewClient(nil, model.Project.MlflowTrackingUrl)
+	mlflowClient := c.MlflowClientFactory.NewClient(nil, model.Project.MlflowTrackingUrl)
```

Comment:
I think it's better to make `VersionsController` hold reference to `mlflowClient` directly. 
To give you some context, this code
```
mlflowClient := mlflow.NewClient(nil, model.Project.MlflowTrackingUrl)
```
was intended to for multiple MLFlow deployments. I don't think it's the case and we always have only 1 MLflow per merlin instance. Thus, it's better to instantiate the mlflow client during start up and store the reference in `VersionsController`

Reply:
Do we want to break this compatibility? Because right now its reading the value from databases.

Reply:
Also, are we pulling this out into a configuration file?